bundles/org.smarthomej.binding.telenot/src/main/java/org/smarthomej/binding/telenot/internal/TelenotDiscoveryService.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2021 Contributors to the SmartHome/J project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.smarthomej.binding.telenot.internal;
+
+import static java.util.Map.entry;
+import static org.smarthomej.binding.telenot.internal.TelenotBindingConstants.*;
+
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.smarthomej.binding.telenot.internal.handler.TelenotBridgeHandler;
+
+/**
+ * The {@link TelenotDiscoveryService} handles discovery of devices as they are identified by the bridge handler.
+ *
+ * @author Ronny Grun - Initial contribution
+ */
+@NonNullByDefault
+public class TelenotDiscoveryService extends AbstractDiscoveryService implements ThingHandlerService {
+
+    private final Logger logger = LoggerFactory.getLogger(TelenotDiscoveryService.class);
+
+    private @Nullable TelenotBridgeHandler bridgeHandler;
+    private @Nullable ThingUID bridgeUID;
+
+    public TelenotDiscoveryService() {
+        super(DISCOVERABLE_DEVICE_TYPE_UIDS, 0, false);
+    }
+
```

Comment:
```suggestion
    public TelenotDiscoveryService() {
        super(DISCOVERABLE_DEVICE_TYPE_UIDS, 0, true);
    }

     @Override
    protected void startBackgroundDiscovery() {
        scheduler.schedule(() -> {
            startScan();
        }, 10, TimeUnit.SECONDS);
    }
```

This will scan every ten seconds. You can adjust to whatever you feel is useful. I think 10-30s is a good compromise between excessive scanning and too long delays.